R/ttestindependentsamples.R

```diff
@@ -631,6 +632,97 @@ ttestIndependentMainTableRow <- function(variable, dataset, test, testStat, effS
+  if (!options$descriptivesPlotsTwo)
+    return()
+  .ttestDescriptivesContainer(jaspResults, options)
+  container <- jaspResults[["ttestDescriptives"]]
+  if (is.null(container[["plotsTwo"]])) {
+    subcontainer <- createJaspContainer(gettext("Bar Plots"), dependencies = c("descriptivesPlotsTwo",
+                                                                               "descriptivesPlotsTwoConfidenceIntervalField",
+                                                                               "errorBarType",
+                                                                               "zeroFix"))
+    subcontainer$position <- 6
+    container[["plotsTwo"]] <- subcontainer
+  } else {
+    subcontainer <- container[["plotsTwo"]]
+  }
+
+  for (variable in options$variables) {
+    if (!is.null(subcontainer[[variable]]))
+      next
+    descriptivesPlotTwo <- createJaspPlot(title = variable, width = 480, height = 320)
+    descriptivesPlotTwo$dependOn(optionContainsValue = list(variables = variable))
+    subcontainer[[variable]] <- descriptivesPlotTwo
+    if (ready) {
+      p <- try(.ttestIndependentDescriptivesPlotTwoFill(dataset, options, variable))
+      if (isTryError(p))
+        descriptivesPlotTwo$setError(.extractErrorMessage(p))
+      else
+        descriptivesPlotTwo$plotObject <- p
+    }
+  }
+  return()
+}
+
+.ttestIndependentDescriptivesPlotTwoFill <- function(dataset, options, variable) {
+  groups <- options$groupingVariable
+
+  errors <- .hasErrors(dataset,
+                       message = 'short',
+                       type = c('observations', 'variance', 'infinity'),
+                       all.target = variable,
+                       observations.amount = '< 2',
+                       observations.grouping = groups)
+
+  if (!identical(errors, FALSE))
+    stop(errors$message)
+
+  dataset <- na.omit(dataset[, c(groups, variable)])
+  ci <- options$descriptivesPlotsTwoConfidenceIntervalField
+
+  if (options$errorBarType == "descriptivesPlotsTwoConfidenceInterval") {
+    summaryStat <- summarySE(as.data.frame(dataset),
+                             measurevar = variable,
+                             groupvars = groups,
+                             conf.interval = ci, na.rm = TRUE, .drop = FALSE)
+  } else if (options$errorBarType == "standardError") {
+    summaryStat <- summarySE(as.data.frame(dataset),
+                             measurevar = variable,
+                             groupvars = groups,
+                             conf.interval = ci, na.rm = TRUE, .drop = FALSE,
+                             errorBarType = "se")
+  }
+
+  colnames(summaryStat)[which(colnames(summaryStat) == variable)] <- "dependent"
+  colnames(summaryStat)[which(colnames(summaryStat) == groups)]   <- "groupingVariable"
+
+  ciPos <- c(summaryStat$ciLower, summaryStat$ciUpper)
+
+  if (options$zeroFix) {
+    yBreaks <- pretty(c(0, ciPos))
+  } else {
+    yBreaks <- pretty(ciPos)
+  }
```

Comment:
```suggestion
  yBreaks <- jaspGraphs::getPrettyAxisBreaks(if (options[["zeroFix"]]) c(0, ciPos) else ciPos)
```
Two things. 

1. don't repeat yourself. It's fine to do 
```r
yBreaks <- if (condition) {
  stuff
} else {
  other stuff
}
```
instead of repeating `yBreaks <-` twice. You can also put the if inside the function, like in the suggestion.

2. `[[` is preferred over `$`. The only difference is that `$` does partial matching while `[[` does not. For example:
```r
lst <- list(apple = 1)
lst$a
[1] 1
lst[["a"]]
NULL
```
This behavior has bitten people in the past, and generally `[[` is safer. Yes, I know we don't do this consistently everywhere in jasp, but when I see it in new code I point it out.